### PR TITLE
Actions: fix tests that don't like repos not called "ardupilot"

### PR DESCRIPTION
**Pull request by IamPete1**
A few of the git safe directory calls assume the repo is called `ardupilot`, if your fork happens not to be called that then CI fails. GitHub provides us with a environmental variable to get the correct directory which we already use in most places. The tricky one is cygwin which uses the repo name, but on a different path, so in that case I have to some messing about to extract the repo name from `GITHUB_REPOSITORY` and `GITHUB_REPOSITORY_OWNER` (see https://tom-gallacher.co.uk/articles/how-to-github-actions-repo-name/)

This also ups the timeout for the environment install scripts from 45 mins to 60 mins. These seem to run a bit slower outside of the main repo, I'm not sure why.

### CODE REVIEW

File: .github/workflows/test_environment.yml [hunk 1]
```diff
@@ -79,7 +79,7 @@ jobs:
         with:
             submodules: 'recursive'
       - name: test install environment ${{matrix.os}}.${{matrix.name}}
-        timeout-minutes: 45
+        timeout-minutes: 60
```
Comment (khancyr) [inline .github/workflows/test_environment.yml:82]: why raising to 60 ? it should be a bit less 30min without cache . But not blocking

**Review by khancyr — APPROVED** [review]
LTGM

truely a git annoyance ...

**Comment by khancyr** [comment]
for the question why it is faster on ardupilot, I think we got an upgraded plan as FOSS project an have better machine for CI.

**Comment by khancyr** [comment]
We can merge without waiting environement tests to pass as they are failing due to some setuptools issues, so not related

**Comment by rmackay9** [comment]
My repo is one of these actually

**Review by tridge — APPROVED** [review]

**Comment by tridge** [comment]
@khancyr can you do a PR against this PR that fixes the tests so env install works again?

**Comment by peterbarker** [comment]
The Jammy environment install was fixed in https://github.com/ArduPilot/ardupilot/pull/27609

Focal is still outstanding

**Pull request merged**